### PR TITLE
Added PointerPressed and PointerReleased events to DataGrid header

### PR DESCRIPTION
**Pull request by ctacke**
## What does the pull request do?

Adds the ability to capture pointer/mouse events at the application for the DataGrid header.

## What is the current behavior?

Currently mouse events trigger sorting when sort is enabled, but do nothing when sort is disabled

## What is the updated/expected behavior with this PR?

An event handler can be attached to the pointer events on a header
Example:
```
var column = new DataGridTextColumn
{
    Header = c.Key,
    Binding = new Binding($"Column{i}")
};
column.HeaderPointerReleased += OnHeaderPointerReleased;
dataGrid.Columns.Add(column);
...
private void OnHeaderPointerReleased(object? sender, Avalonia.Input.PointerReleasedEventArgs e)
{
}

```
## How was the solution implemented (if it's not obvious)?

It is obvious

## Checklist

- [ ] Added unit tests (if possible)?
- [ X ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
No breaking changes

## Obsoletions / Deprecations

No obsoletions

## Fixed issues

No issue reported

### CODE REVIEW

**Comment by Artfunkel** [comment]
General observation: in WPF this (and much more) can be achieved with `EventSetter` in a style targeting column headers. I really miss this feature in Avalonia.

**Comment by TomEdwardsEnscape** [comment]
General observation: in WPF this (and much more) can be achieved with `EventSetter` in a style targeting column headers. I really miss this feature in Avalonia.

**Comment by ctacke** [comment]
@microsoft-github-policy-service agree

**Comment by maxkatz6** [comment]
@timunie yeah, it can be discussed

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Controls.DataGrid/Avalonia.Controls.DataGrid.csproj [hunk 1]
```diff
@@ -1,20 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netstandard2.0</TargetFrameworks>
```
Comment (maxkatz6) [inline src/Avalonia.Controls.DataGrid/Avalonia.Controls.DataGrid.csproj:3]: It looks like unwanted changes were pushed with the latest rebase

**Review by maxkatz6 — CHANGES_REQUESTED** [review]

**Comment by ctacke** [comment]
@dotnet-policy-service agree

File: src/Avalonia.Controls.DataGrid/Avalonia.Controls.DataGrid.csproj [hunk 2]
```diff
@@ -3,17 +3,10 @@
...
-    <ProjectReference Include="..\Markup\Avalonia.Markup.Xaml\Avalonia.Markup.Xaml.csproj" />
-    <ProjectReference Include="..\Markup\Avalonia.Markup\Avalonia.Markup.csproj" />    
-    <ProjectReference Include="..\Avalonia.Controls\Avalonia.Controls.csproj" />
+    <PackageReference Include="Avalonia" Version="11.0.3" />
     <!-- Compatibility with old apps -->
     <EmbeddedResource Include="Themes\**\*.xaml" />
   </ItemGroup>
-  <Import Project="..\..\build\EmbedXaml.props" />
-  <Import Project="..\..\build\BuildTargets.targets" />
-  <Import Project="..\..\build\DevAnalyzers.props" />
```
Comment (maxkatz6) [inline src/Avalonia.Controls.DataGrid/Avalonia.Controls.DataGrid.csproj:16]: Please revert all changes in this project file. It seems to be a debug changes, not required for this PR.

**Review by maxkatz6 — APPROVED** [review]
LGTM

**Pull request merged**